contracts/SimpleYieldEulerLever.sol

```diff
@@ -0,0 +1,395 @@
+    //     |                                                    | sell y to get underlying to payback the flashloan        |            |            |         |
+    //     |                                                    |--------------------------------------------------------------------------------------------->|
+    //     |                                                    |                                                          |            |            |         |
+    //     |                Transfer the vault back to the user |                                                          |            |            |         |
+    //     |<---------------------------------------------------|                                                          |            |            |         |
+    //     |                                                    |                                                          |            |            |         |
+    function invest(
+        bytes6 seriesId,
+        uint128 baseAmount,
+        uint128 borrowAmount,
+        uint128 minCollateral
+    ) external returns (bytes12 vaultId) {
+        // Transfer the tokens from user based on the ilk
+        IERC20(address(eToken)).safeTransferFrom(
+            msg.sender,
+            address(this),
+            baseAmount
+        );
+
+        // Build vault
+        (vaultId, ) = ladle.build(seriesId, ILKID, 0);
+
+        // Encode data of
+        // OperationType    1 byte      [0:1]
+        // seriesId         6 bytes     [1:7]
+        // vaultId          12 bytes    [7:19]
+        // baseAmount       16 bytes    [19:35]
+        // minCollateral    16 bytes    [35:51]
+        bytes memory data = bytes.concat(
+            bytes1(uint8(uint256(Operation.BORROW))),
+            seriesId,
+            vaultId,
+            bytes16(baseAmount),
+            bytes16(minCollateral)
+        );
+
+        bool success = assetJoin.flashLoan(
+            this, // Loan Receiver
+            asset, // Loan Token
+            borrowAmount, // Loan Amount
+            data
+        );
+
+        if (!success) revert FlashLoanFailure();
+        giver.give(vaultId, msg.sender);
+    }
+
+    /// @notice Called by a flash lender, which will be the join of the asset underlying the eToken
+    ///     The primary purpose is to check conditions
+    ///     and route to the correct internal function.
+    ///
+    ///     This function reverts if not called through a flashloan initiated
+    ///     by this contract.
+    /// @param initiator The initator of the flash loan, must be `address(this)`.
+    /// @param borrowAmount The amount of fyTokens received.
+    /// @param fee The fee that is subtracted in addition to the borrowed
+    ///     amount when repaying.
+    /// @param data The data we encoded for the functions. Here, we only check
+    ///     the first byte for the router.
+    function onFlashLoan(
+        address initiator,
```

Comment:
```suggestion
        address,
```
I think that we don't need to check the initiator, since the lever doesn't keep any vaults between transactions. If someone would call `onFlashLoan` with malicious data, there would be no vault that they could do anything to anyway.

Please keep this text as a comment explaining why there isn't a verification of the caller on `onFlashLoan`.